services/table-foreign-keys-analyzer.js

```diff
@@ -8,19 +8,41 @@ function TableForeignKeysAnalyzer(databaseConnection, schema) {
     switch (queryInterface.sequelize.options.dialect) {
       case 'postgres':
         query = `
-          SELECT
-            tc.constraint_name,
-            tc.table_name,
-            kcu.column_name,
-            ccu.table_name AS foreign_table_name,
-            ccu.column_name AS foreign_column_name
-          FROM information_schema.table_constraints AS tc
-          JOIN information_schema.key_column_usage AS kcu
-            ON tc.constraint_name = kcu.constraint_name
-          JOIN information_schema.constraint_column_usage AS ccu
-            ON ccu.constraint_name = tc.constraint_name
-          WHERE constraint_type = 'FOREIGN KEY'
-            AND tc.table_name=:table;`;
+        SELECT 
+          table_constraints.constraint_name,
+          table_constraints.table_name,
+          table_constraints.constraint_type AS column_type,
+          key_column_usage.column_name,
+          constraint_column_usage.table_name AS foreign_table_name,
+          constraint_column_usage.column_name AS foreign_column_name,
+          json_agg(uidx.unique_indexes) filter (where uidx.unique_indexes is not null) AS unique_indexes
+        FROM information_schema.table_constraints AS table_constraints
+        JOIN information_schema.key_column_usage AS key_column_usage
+          ON table_constraints.constraint_name = key_column_usage.constraint_name
+        JOIN information_schema.constraint_column_usage AS constraint_column_usage
+          ON constraint_column_usage.constraint_name = table_constraints.constraint_name
+        FULL OUTER JOIN (
+          SELECT pg_index.indexrelid::regclass AS index_name,
+              pg_class1.relname AS table_name,
+              json_agg(DISTINCT pg_attribute.attname) AS unique_indexes
+          FROM
+              pg_class AS pg_class1,
+              pg_class AS pg_class2,
+              pg_index,
+              pg_attribute
+          WHERE
+              pg_class1.oid = pg_index.indrelid
+              AND pg_class2.oid = pg_index.indexrelid
+              AND pg_attribute.attrelid = pg_class1.oid
+              AND pg_attribute.attnum = ANY(pg_index.indkey)
+              AND not pg_index.indisprimary
+              AND pg_index.indisunique
+              AND pg_class1.relkind = 'r'
+              AND not pg_class1.relname like 'pg%'
```

Comment:
A little explanation would be appreciated in a comment. Always think about people reading after you